chain/network/src/store/mod.rs

```diff
@@ -0,0 +1,185 @@
+/// All transactions should be implemented within this module,
+/// in particular schema::StoreUpdate is not exported.
+use near_network_primitives::types::{Edge, KnownPeerState};
+use near_primitives::network::{AnnounceAccount, PeerId};
+use near_primitives::types::AccountId;
+use std::collections::HashSet;
+use tracing::debug;
+
+mod schema;
+
+pub use schema::Error;
+
+/// Store allows for performing synchronous atomic operations on the DB.
+/// In particular it doesn't implement Clone and requires &mut self for
+/// methods writing to the DB.
+pub struct Store(schema::Store);
+
+impl Store {
+    pub fn new(s: near_store::Store) -> Self {
+        Self(schema::Store::new(s))
+    }
+}
+
+/// Everytime a group of peers becomes unreachable at the same time; We store edges belonging to
+/// them in components. We remove all of those edges from memory, and save them to database,
+/// If any of them become reachable again, we re-add whole component.
+///
+/// To store components, we have following column in the DB.
+/// DBCol::LastComponentNonce -> stores component_nonce: u64, which is the lowest nonce that
+///                          hasn't been used yet. If new component gets created it will use
+///                          this nonce.
+/// DBCol::ComponentEdges     -> Mapping from `component_nonce` to list of edges
+/// DBCol::PeerComponent      -> Mapping from `peer_id` to last component nonce if there
+///                          exists one it belongs to.
+impl Store {
+    pub fn set_account_announcement(
+        &mut self,
+        account_id: &AccountId,
+        aa: &AnnounceAccount,
+    ) -> Result<(), Error> {
+        let mut update = self.0.new_update();
+        update.set::<schema::AccountAnnouncements>(account_id, aa);
+        update.commit()
+    }
+
+    pub fn get_account_announcement(
+        &self,
+        account_id: &AccountId,
+    ) -> Result<Option<AnnounceAccount>, Error> {
+        self.0.get::<schema::AccountAnnouncements>(account_id)
+    }
+
+    pub fn push_component(
+        &mut self,
+        peers: &HashSet<PeerId>,
+        edges: &Vec<Edge>,
+    ) -> Result<(), Error> {
+        debug!(target: "network", "push_component: We are going to remove the following peers: {}", peers.len());
+        let component = self.0.get::<schema::LastComponentNonce>(&())?.unwrap_or(0) + 1;
+        let mut update = self.0.new_update();
+        update.set::<schema::LastComponentNonce>(&(), &component);
+        update.set::<schema::ComponentEdges>(&component, &edges);
+        for peer_id in peers {
+            update.set::<schema::PeerComponent>(peer_id, &component);
+        }
+        update.commit()
+    }
+
+    pub fn pop_component(&mut self, peer_id: &PeerId) -> Result<Vec<Edge>, Error> {
+        // Fetch the component assigned to the peer.
+        let component = match self.0.get::<schema::PeerComponent>(peer_id)? {
+            Some(c) => c,
+            _ => return Ok(vec![]),
```

Comment:
```suggestion
            None => return Ok(vec![]),
```

`_` often are a smell. It doesn't matter in this case, but it's not hard to make the match exhaustive anyway.

Reply:
done